src/utils/ParseRouteUtils.js

```diff
@@ -532,8 +535,135 @@ function parseRoute(resp: Object, destinationName: string) {
+    endCoords: { lat: 42.45662677611252, long: -76.47693624444763 },
+    startName: 'Risley Hall - Shelter',
+    endName: 'Ithaca Commons - Seneca St',
+    boundingBox:
+     { minLat: 42.441596,
+       minLong: -76.490387,
+       maxLat: 42.456818,
+       maxLong: -76.471642 },
+    travalDistance: 1231,
+    totalDuration: 12,
+    numberOfTransfers: 1,
+    routeSummary: [...],
+  },
+ ...
+  ]
+ * @param resp
+ * @param destinationName
+ * @returns {Promise<Array>}
+ */
+async function parseDetailedRoute(resp: Object, destinationName: string) {
+    const paths = await parseRoute(resp, destinationName);
+    
+    return paths.map((currPath) => {
+        const {
+            arrivalTime,
+            departureTime,
+            directions,
+            endCoords,
+            startCoords,
+        } = currPath;
+
+        let travelDistance = geolib.getDistance(
+            { latitude: startCoords.lat, longitude: startCoords.long },
+            { latitude: endCoords.lat, longitude: endCoords.long }
+        );
+
+        travelDistance *= METERS_TO_MILES;
+        currPath.travelDistance = travelDistance;
+
+        let totalDuration = new Date(arrivalTime) - new Date(departureTime);
+        totalDuration /= ONE_MINUTE_IN_MILLISECONDS;
+        currPath.totalDuration = Math.round(totalDuration);
+
+        currPath.startName = currPath.directions[0].name;
+        currPath.endName = destinationName;
+
+        // routeSummary
+
+        /* routeSummaryElement {
+            stopName: String,
+            direction: RouteSummaryDirection?,
+            shouldStayOnBus: Bool,
+        }
+        */
+
+        /* routeSummaryDirection {
+            type: null || 'walk' || 'bus',
+            routeNumber: null || Int,
+        }
+        */
+
+        const routeSummary = [];
+
+        let walkOnlyRoute = true;
+        for (let i = 0; i < directions.length; i++) {
+            walkOnlyRoute = walkOnlyRoute && (directions[i].type === 'walk');
+        }
+
+        // if walk only route, the first element of routeSummary will be 'walk'
+        if (walkOnlyRoute) {
+            routeSummary.push({
+                stopName: directions[0].name,
+                direction: { type: 'walk', routeNumber: null },
+                shouldStayOnBus: false,
+            });
+            routeSummary.push({
+                stopName: directions[directions.length - 1].name,
+                direction: { type: null, routeNumber: null },
+                shouldStayOnBus: false,
+            });
+        } else {
+            let indexOfFirstBus = -1;
+            for (let i = 0; i < directions.length; i++) {
+                if (directions[i].type === 'depart') {
+                    indexOfFirstBus = i;
+                    break;
+                }
+            }
+
+            for (let i = indexOfFirstBus; i < directions.length; i++) {
+                const newRouteSummary = { stopName: directions[i].name, shouldStayOnBus: false };
```

Comment:
Noticed that `shouldStayOnBus` is always set to `false`. Is there somewhere else in the code that you use the results from GraphHopper to check if you `shouldStayOnBus`? B/c from what I understand GraphHopper tells us that.